src/java.base/share/classes/java/lang/template/StringTemplate.java

```diff
@@ -0,0 +1,368 @@
+     */
+    List<String> fragments();
+
+    /**
+     * Returns an immutable list of embedded expression results. In the example:
+     * {@snippet :
+     * StringTemplate st = RAW."\{x} + \{y} = \{x + y}";
+     * List<Object> values = st.values(); // @highlight substring="values()"
+     * }
+     * <code>values</code> will be equivalent to <code>List.of(x, y, x + y)</code>
+     *
+     * @return list of expression values
+     *
+     * @implSpec The list returned is immutable.
+     */
+    List<Object> values();
+
+    /**
+     * {@return the interpolation of the StringTemplate}
+     */
+    default String interpolate() {
+        return TemplateRuntime.interpolate(fragments(), values());
+    }
+
+    /**
+     * Returns the result of applying the specified processor to this {@link StringTemplate}.
+     * This method can be used as an alternative to string template expressions. For example,
+     * {@snippet :
+     * String result1 = STR."\{x} + \{y} = \{x + y}";
+     * String result2 = RAW."\{x} + \{y} = \{x + y}".process(STR); // @highlight substring="process"
+     * }
+     * produces an equivalent result for both {@code result1} and {@code result2}.
+     *
+     * @param processor the {@link ValidatingProcessor} instance to process
+     *
+     * @param <R>  Processor's process result type.
+     * @param <E>  Exception thrown type.
+     *
+     * @return constructed object of type R
+     *
+     * @throws E exception thrown by the template processor when validation fails
+     * @throws NullPointerException if processor is null
+     *
+     * @implNote The default implementation simply invokes the processor's process
+     * method {@code processor.process(this)}.
+     */
+    default <R, E extends Throwable> R process(ValidatingProcessor<R, E> processor) throws E {
+        Objects.requireNonNull(processor, "processor should not be null");
+
+        return processor.process(this);
+    }
+
+    /**
+     * Return the types of this {@link StringTemplate StringTemplate's} values.
+     *
+     * @return list of value types
+     *
+     * @implNote The default method determines if the {@link StringTemplate}
+     * was synthesized by the compiler, then the types are precisely those of the
+     * embedded expressions, otherwise this method returns the values list types.
+     */
+    default public List<Class<?>> valueTypes() {
+        return TemplateRuntime.valueTypes(this);
+    }
+
+    /**
+     * Return {@link MethodHandle MethodHandles} to access this
+     * {@link StringTemplate StringTemplate's} values.
+     *
+     * @return list of value types
+     *
+     * @implNote The default method determines if the {@link java.lang.template.StringTemplate}
+     * was synthesized by the compiler, then the MethodHandles are precisely those of the
+     * embedded expressions fields, otherwise this method returns getters for the values list.
+     */
+    default public List<MethodHandle> valueGetters() {
```

Comment:
I think i prefer the term accessors instead of getters

Reply:
Changing